demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -407,6 +480,146 @@ static TransportSocketStatus_t prvConnectToServerWithBackoffRetries( NetworkCont
+    uint32_t ulTopicCount = 0U;
+    BaseType_t xStatus = pdFAIL;
+
+    assert( pxMqttContext != NULL );
+    assert( pcTopicFilter != NULL );
+    assert( usTopicFilterLength > 0 );
+
+    /* Some fields not used so start with everything at 0. */
+    ( void ) memset( ( void * ) &xMQTTSubscription, 0x00, sizeof( xMQTTSubscription ) );
+
+    /* Initialize the status of the subscription. */
+    xTopicFilterContext.pcTopicFilter = pcTopicFilter;
+    xTopicFilterContext.xSubAckStatus = MQTTSubAckFailure;
+
+    /* Get a unique packet id. */
+    globalSubscribePacketIdentifier = MQTT_GetPacketId( pxMqttContext );
+
+    /* Subscribe to the pcTopicFilter topic filter. This example subscribes to
+     * only one topic and uses QoS1. */
+    xMQTTSubscription.qos = MQTTQoS1;
+    xMQTTSubscription.pTopicFilter = pcTopicFilter;
+    xMQTTSubscription.topicFilterLength = usTopicFilterLength;
+
+    /* Initialize retry attempts and interval. */
+    BackoffAlgorithm_InitializeParams( &xRetryParams,
+                                       SUBSCRIBE_RETRY_BACKOFF_BASE_MS,
+                                       SUBSCRIBE_RETRY_MAX_BACKOFF_DELAY_MS,
+                                       SUBSCRIBE_RETRY_MAX_ATTEMPTS );
+
+    do
+    {
+        /* The client is now connected to the broker. Subscribe to the topic
+         * as specified in pcTopicFilter at the top of this file by sending a
+         * subscribe packet then waiting for a subscribe acknowledgment (SUBACK).
+         * This client will then publish to the same topic it subscribed to, so it
+         * will expect all the messages it sends to the broker to be sent back to it
+         * from the broker. This demo uses QOS0 in Subscribe, therefore, the Publish
+         * messages received from the broker will have QOS0. */
+        LogInfo( ( "Attempt to subscribe to the MQTT topic %s.", pcTopicFilter ) );
+        xResult = MQTT_Subscribe( pxMqttContext,
+                                  &xMQTTSubscription,
+                                  sizeof( xMQTTSubscription ) / sizeof( MQTTSubscribeInfo_t ),
+                                  globalSubscribePacketIdentifier );
+
+        if( xResult != MQTTSuccess )
+        {
+            LogError( ( "Failed to SUBSCRIBE to MQTT topic %s. Error=%s",
+                        pcTopicFilter, usTopicFilterLength ) );
+        }
+        else
+        {
+            xStatus = pdPASS;
+            LogInfo( ( "SUBSCRIBE sent for topic %s to broker.", pcTopicFilter ) );
+
+            /* Process incoming packet from the broker. After sending the subscribe, the
+             * client may receive a publish before it receives a subscribe ack. Therefore,
+             * call generic incoming packet processing function. Since this demo is
+             * subscribing to the topic to which no one is publishing, probability of
+             * receiving Publish message before subscribe ack is zero; but application
```

Comment:
Is the statement about no one is publishing to the topic correct?

Reply:
I don't think. Like for Shadow, a delta may be received from another device. I've removed the statement about probability being 0.